test/gtest/common/test.cc

```diff
@@ -216,6 +221,23 @@ test_base::wrap_errors_logger(const char *file, unsigned line, const char *funct
     return UCS_LOG_FUNC_RC_CONTINUE;
 }
 
+ucs_log_func_rc_t
+test_base::detect_error_logger(const char *file, unsigned line, const char *function,
+                               ucs_log_level_t level, const char *message, va_list ap)
+{
+    if (level == UCS_LOG_LEVEL_ERROR) {
+        pthread_mutex_lock(&m_logger_mutex);
```

Comment:
why need a mutex here?

Reply:
why is it needed in wrap_errors_logger()? isn't this the same?

Reply:
it protects m_errors there